fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckService.kt

```diff
@@ -216,3 +240,28 @@ enum class AnalysisStatusDto {
      */
     OTHER,
 }
+
+/**
+ * Enum that represents the status returned from reporting a product is back in stock.
+ */
+enum class ReportBackInStockStatusDto {
+    /**
+     * Created reported.
+     */
+    REPORT_CREATED,
+
+    /**
+     * Product is already reported to be back in stock.
+     */
+    ALREADY_REPORTED,
+
+    /**
+     * Product was not actually marked as deleted.
+     */
+    NOT_DELETED,
+
+    /**
+     * Any other status.
+     */
+    OTHER,
```

Comment:
Since this API has 3 states and we handle them specifically, we don't need to define this `Other` state IMO

Reply:
Sure, removed `OTHER` and set the default for `asEnumOrDefault` to `NOT_DELETED`, although I'm not sure if that's the best default between the 3.

Reply:
We don't need a default since API returns one of the 3 strings/enum values. If it's something else, then the service will return null which is handled in the middleware.